### DISCUSSION TITLE
how to "turn off" the term presence lunr setting for queries that start with a hyphen `-`

### DISCUSSION THREAD

**Discussion by flynneva** [squidfunk/mkdocs-material#3163] [Q&A] [answered]
After doing some digging on [why search queries that start with a hyphen](https://lunrjs.com/guides/searching.html) (see the `Term Presence` section) dont seem to return the expected results I was wondering how to "turn off" this feature of lunr so that when a user queries `--some-argument` they get the expected result?

I noticed that [on the mkdocs site](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/?q=--force) querying for `--force` returns the expected results so I am sure it is possible!

**Comment by flynneva** [comment, marked as answer]
so I was able to semi-figure it out through a hacky way using [the custom overrides feature of mkdocs](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/#customization) to remove the leading `-` if present before submitting the query to lunr. This has a drawback of now removing the "exclude" feature that leading `-` are supposed to have...but for our use case _I think_ this solution might work.

here is what i added to our `overrides/main.html`:

```html
{% block config %}
  {{ super() }}
  <script>
    var __search = {
      transform: function(query) {
        return query
          .split(/"([^"]+)"/g) 
            .map((terms, index) => index & 1
              ? terms.replace(/^\b|^(?![^\x00-\x7F]|$)|\s+/g, " +")
              : terms
            )
            .join("")
            .replace(/"|(^\-+)|(?:^|\s+)[*+:^~]+(?=\s+|$)/g, "")
          .trim()
      }
    }
  </script>
{% endblock %}
```

**Reply by squidfunk** [reply]
> I noticed that on the mkdocs site querying for --force returns the expected results so I am sure it is possible!

This is because `-` is used as a separator, so `--force` is tokenized into `force`, which is why it's discoverable. I don't think it's possible to craft a regular expression that detects `-` as a separator in `foo-bar`, but leaves it as-is in `--foo`, but I may be mistaken.

**Reply by flynneva** [reply]
@squidfunk I had already adjusted our separator tokenizer regex to only include a `-` if it is not followed by a non-whitespace character `\-(?!\S)`  (I actually do it the opposite, so check if it is whitespace and include it `\-(?=\s)`) very similar to how you recommended to [find version numbers](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/#tokenizer-lookahead) in your docs.

I guess an alternative solution would be to switch the above html code to only replace the leading hypens if there are exactly two - that way the original functionality will still be there for excluding terms from results.  This approach would work for many command line args but not all I guess.
